### PR TITLE
feat(react): implement icon button variants

### LINKED ISSUE #177: Implement Icon Button variants
[UXPin (Pattern Library)](https://preview.uxpin.com/5ca5c040146a4ebb4536c10998dd84d1199e5d79#/pages/134488442/specification/sitemap)

cauldron icon button patterns and variants

### PR DESCRIPTION
**Pull request by scurker**
closes #177 

This should bring `<IconButton>` in pretty close alignment with `<Button>`.

### CODE REVIEW

File: packages/react/src/components/IconButton/index.tsx [hunk 1]
```diff
@@ -23,18 +24,25 @@ const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
       label,
       tooltipPlacement = 'auto',
       className,
+      variant = 'secondary',
       ...other
     }: IconButtonProps,
-    ref
+    ref: any
```
Comment (schne324) [inline packages/react/src/components/IconButton/index.tsx:30]: ❗ add a prop ref type here

**Review by schne324 — CHANGES_REQUESTED** [review]
everything looks good to me aside from [my ref type comment](https://github.com/dequelabs/cauldron/pull/181/files#r594589043)

File: packages/react/src/components/IconButton/index.tsx (hunk 1 above)
Comment (scurker) [reply packages/react/src/components/IconButton/index.tsx]: Sorry, not sure why that was still there. It should be an inferred type.

File: packages/react/src/components/IconButton/index.tsx [hunk 2]
```diff
@@ -56,7 +64,9 @@ const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
 IconButton.propTypes = {
   icon: PropTypes.string.isRequired,
   label: PropTypes.string.isRequired,
-  tooltipPlacement: PropTypes.any
+  tooltipPlayemnt: PropTypes.string,
```
Comment (okry) [inline packages/react/src/components/IconButton/index.tsx:67]: typo: should be `tooltipPlacement`

**Review by okry — DISMISSED** [review]

File: packages/react/src/components/IconButton/index.tsx (hunk 2 above)
Comment (scurker) [reply packages/react/src/components/IconButton/index.tsx:67]: words are hard

**Review by schne324 — APPROVED** [review]

**Pull request merged**